### PR TITLE
Clean up and Format System.Collections

### PR DESCRIPTION
**Pull request by ctyar**
I need some feedback about the amount of cleanup that needs to be done
I did the basic things that asked in main issue(https://github.com/dotnet/corefx/issues/5055) minus renaming the fields, this comment kinda scared me:
> Non-compliant field names
We need to be careful about renaming these, we've hit issues in the past where Debugger components were relying on the exact names of these fields to not change.

The documentations says existing style in the files takes precedence so I'm not sure how much change is ok, Is this enough?

PS: I haven't touched the tests yet, I wanted to be sure this is the right approach

### CODE REVIEW

File: src/System.Collections/src/System/Collections/Generic/SortedSet.cs [hunk 1]
```diff
@@ -333,8 +328,9 @@ object ICollection.SyncRoot
         #region Subclass helpers
 
         //virtual function for subclass that needs to update count
-        internal virtual void VersionCheck() { }
-
+        internal virtual void VersionCheck()
+        {
+        }
```
Comment (ianhays) [inline src/System.Collections/src/System/Collections/Generic/SortedSet.cs]: FWIW, I'm not a fan of forcing empty brackets to take up their own lines like this. Is this in the style guide?

**Comment by ianhays** [comment]
Thanks @ctyar, these formatting changes look good.  Have you run the [Code formatter tool?](https://github.com/dotnet/codeformatter)

>minus renaming the fields, this comment kinda scared me:

@mellinoe 

>The documentations says existing style in the files takes precedence so I'm not sure how much change is ok, Is this enough?

I don't think that applies when you're doing a sweeping cleanup of the formatting like this - that's more for when you're only making a small change or a change not related to formatting. The above might contradict this, but I'll let Eric define that better since I'm not too familiar with the naming/debugging issue he mentioned.

>PS: I haven't touched the tests yet, I wanted to be sure this is the right approach

Just a heads up, you likely won't have to much to do in the tests. I rewrote them since making that issue.

File: src/System.Collections/src/System/Collections/Generic/HashSet.cs [hunk 2]
```diff
@@ -75,9 +70,9 @@ public class HashSet<T> : ICollection<T>, ISet<T>, IReadOnlyCollection<T>
 
         #region Constructors
 
-        public HashSet()
-            : this(EqualityComparer<T>.Default)
-        { }
+        public HashSet() : this(EqualityComparer<T>.Default)
```
Comment (mellinoe) [inline src/System.Collections/src/System/Collections/Generic/HashSet.cs]: I don't really care for this sort of change. It doesn't really go against our style guide, except for the braces being on the same line.

File: src/System.Collections/src/System/Collections/Generic/SortedList.cs [hunk 3]
```diff
@@ -49,8 +48,8 @@ namespace System.Collections.Generic
     // 
     [DebuggerTypeProxy(typeof(IDictionaryDebugView<,>))]
     [DebuggerDisplay("Count = {Count}")]
-    public class SortedList<TKey, TValue> :
-        IDictionary<TKey, TValue>, System.Collections.IDictionary, IReadOnlyDictionary<TKey, TValue>
+    public class SortedList<TKey, TValue> :
```
Comment (mellinoe) [inline src/System.Collections/src/System/Collections/Generic/SortedList.cs]: Looks like there is extra whitespace at the end of this line.

File: src/System.Collections/src/System/Collections/Generic/SortedSet.cs (hunk 1 above)
Comment (mellinoe) [inline src/System.Collections/src/System/Collections/Generic/SortedSet.cs]: Yeah, I agree on this one. The style guide sort of implies that braces should go on their own line, but it's not really specifically referring to situations like this.

**Comment by mellinoe** [comment]
Most of the changes look good to me. I made a couple of minor comments.

As for renaming non-compliant fields: I think we need to do it on a case-by-case basis. I don't think there is any sort of "master list" that has all of the private fields read by the debugger, so we would need to loop in some debugger team folks to help verify that nothing would be broken by any given rename.

**Comment by mellinoe** [comment]
@cshung may be able to shine some more light on which members are inspected by the debugger.

I see this PR is just coming through now: https://github.com/dotnet/corefx/pull/9853/files

Is that an exhaustive list of fields read by diagnostic tools?

**Comment by cshung** [comment]
@mellinoe Regarding to the question "Is that an exhaustive list of fields read by diagnostic tools?", the answer is unfortunately "yes and no".

My current task is to capture all the fields that are inspected by the debugger in the .NET native debugging in particular. In this case, the answer is "yes", I tried my best to capture as much as I know. However, this work involve reading code into the all the diagnostics tools code base which I am not familiar with, and chances is that I missed something. If I missed anything, we will add as soon as we find them.

There are a bunch of fields in the CoreRT repo as well, for example, we have:
https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/MembersMustExist.AnalyzerData

The answer is clearly no for CoreCLR scenarios. We are planning to capture those as well, but it is not done yet.

I am working on another pull request to document how these fields are used in the debugger. Hope this helps: 
https://github.com/dotnet/corert/pull/1467

**Comment by ctyar** [comment]
Updated the PR per @ianhays , @mellinoe feedback.
Moving on to tests.

File: src/System.Runtime/src/System.Runtime.Forwards.cs [hunk 4]
```diff
@@ -1,3 +1,7 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
```
Comment (ianhays) [inline src/System.Runtime/src/System.Runtime.Forwards.cs]: duplicate header

File: src/Common/src/System/Globalization/IdnMapping.Windows.cs [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 namespace System.Globalization
 {
-    sealed partial class IdnMapping
+    internal sealed partial class IdnMapping
```
Comment (ianhays) [inline src/Common/src/System/Globalization/IdnMapping.Windows.cs]: This change doesn't have anything to do with Collections. I'd say revert it and keep the purpose of this PR singular.

**Comment by ianhays** [comment]
I'd say revert the "	Cleanup with Code formatter tool solution wide" commit. It makes some bad changes and some changes unrelated to System.Collections. The other new commits look good.

**Comment by ctyar** [comment]
@ianhays yeah that was a bad idea, reverted

**Comment by ctyar** [comment]
Is there anything more I need to do or should I just rebase and squash the commits?

**Comment by ianhays** [comment]
@ctyar Nope, nothing else! Just squash it down to one or two commits (your choice) and I'll give it a final look through then merge.

**Comment by ctyar** [comment]
:up: :date:

**Comment by ianhays** [comment]
Great, thanks for the contribution @ctyar!

**Pull request merged**